### PR TITLE
add go basics

### PR DESCRIPTION
**Pull request by jonas27**

### CODE REVIEW

File: server/main.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(exitFail)
+	}
+}
+
+func run(args []string) error {
+	port := flag.String("port", ":8080", "The server port with colon")
+	flag.Parse()
+
+	log.Printf("Server running on port %s\n", *port)
```
Comment (avorima) [inline server/main.go]: log package actually appends a newline to all messages

File: server/main.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+func main() {
+	if err := run(os.Args); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(exitFail)
+	}
+}
+
+func run(args []string) error {
+	port := flag.String("port", ":8080", "The server port with colon")
```
Comment (avorima) [inline server/main.go]: Just being nitpicky here, but isn't a port with a colon actually an address? Just a shorter version of `*:8080`

Comment (jonas27) [reply server/main.go]: yep, thats bad naming from my side. Will change port to addr

Comment (jonas27) [reply server/main.go]: done @avorima

**Review by avorima — APPROVED** [review]

**Pull request merged**